pkg/config/middleware.go

```diff
@@ -120,6 +128,162 @@ func setupRecoveryMiddleware() *negroni.Recovery {
 	return r
 }
 
+type OidcConfiguation struct {
+	JwkURI string `json:"jwks_uri"` // This is the only field we care about for now.
+}
+
+type OidcJwk struct {
+	JWKeyID     string   `json:"kid"` // JWTs will have a "kid" that will match one of these
+	SigningKeys []string `json:"x5c"` // We prefer the x5c since it's easier
+	Exponent    string   `json:"e"`   // The least preferred way is recreating the key through exponent.
+	Modulus     string   `json:"n"`   // The modulus
+}
+
+type OidcJwksConfiguration struct {
+	Jwks []OidcJwk `json:"keys"`
+}
+
+func discoverOidcJwk(tokenKeyID string) *OidcJwk {
```

Comment:
idiomatic go may be better to return the error directly here, so that the error can be unit tested and bubble up to the caller so that they can handle it at their disposal. 

```
func discoverOidcJwk(tokenKeyID string) (*OidcJwk, error) {

        ...
	if err != nil {
		return nil, err
	}
}

```